### DISCUSSION TITLE
Price depending on meta-data

### DISCUSSION THREAD

**Discussion by benoj** [saleor/saleor#6661] [Q&A]
Hi there and happy new year!,

I'm looking to build an a product where it's price depends on metadata properties (e.g. weight of gold). The way I was thinking of doing this was:

1. Add weight of gold as a meta property
2. Job which runs on a CRON which pulls the current price of gold
3. Update the product price dependent upon the weight

Just wondering if there is a better way to achieve this through some internal API's?

Also in the [roadmap](https://saleor.io/open-source/#roadmap) page it references 
```
Multi-currency
This feature will enable support of multi-currencies and individual prices depending on requirements.
```

This sounds potentially like it may actually be something similar.. However I can't find this in the Github project - is this still in the roadmap (and is it what I think it is?)

Thanks,

Ben

**Comment by krzysztofwolski** [comment]
Product prices could be changed directly - you would need to create a Token which would give the script sufficient permissions to edit prices.
You can configure multiple currencies using multichannel. It's available in 3.0 releases.
https://docs.saleor.io/docs/3.0/developer/extending
